### PR TITLE
fix: add screenCap to cached props for diffing during render check

### PR DESCRIPTION
**Pull request by seve**
Added some checks to see if screenCap differs from last attempt to re-render before actually triggering re-render.

Also adds new spatial dataset to example-datasets

These changes prevent the image download flow from download multiple images when pressed a single time while the spatial feature flag is enabled

#823

### CODE REVIEW

File: client/src/components/graph/graph.tsx [hunk 1]
```diff
@@ -863,9 +866,15 @@ class Graph extends React.Component<GraphProps, GraphState> {
...
-      asyncProps;
-    const { screenCap } = this.props;
+    const {
+      positions,
+      colors,
+      flags,
+      height,
+      width,
+      imageUnderlay,
+      screenCap,
```
Comment (tihuan) [inline client/src/components/graph/graph.tsx:876]: just wanted to double check, `screenCap` was coming from `this.props`, but now we're getting it from `asyncProps`. Is that intentional? Thanks!

File: example-dataset/super-cool-spatial.cxg/Xc/__commits/__1708035189525_1708035193898_3917086ff41e413989ed3d822fc83b1f_18.wrt
Comment (tihuan) [inline example-dataset/super-cool-spatial.cxg/Xc/__commits/__1708035189525_1708035193898_3917086ff41e413989ed3d822fc83b1f_18.wrt:1]: these files can all be deleted?

Comment (tihuan) [reply example-dataset/super-cool-spatial.cxg/Xc/__commits/__1708035189525_1708035193898_3917086ff41e413989ed3d822fc83b1f_18.wrt:1]: Ah nvm, this is the new spatial dataset. Thanks for adding this, Seve!

**Comment by tihuan** [comment]
@seve can you help putting down a description of the bug, so the reviewers have more context please? Thank you!

File: client/src/components/graph/graph.tsx (hunk 1 above)
Comment (seve) [reply client/src/components/graph/graph.tsx:876]: it's not necessary to get it from asyncProps here, both places are correct. The reason we're adding screenCap to watchProps is to piggyback on the prop caching.

**Comment by tihuan** [comment]
@seve PTAL this approach! Would love to see what you think 🙏 

https://github.com/chanzuckerberg/single-cell-explorer/compare/thuang-fix-double-download?expand=1

Thank you!

**Review by tihuan — APPROVED** [review]
LGTM!! Thanks for the great fix, @seve 🚀 👏 !!

**Pull request merged**